syntax = "proto3";

package ProtoTransport;

message Edge {
    uint32 from = 1;
    uint32 to = 2;
    double time = 3;
}

message EdgeInfo {
    enum EdgeType {
        WaitBus = 0;
        RideBus = 1;
        WalkToCompany = 2;
    };
    EdgeType type = 1;
    string item_name = 2;
    int32 span_count = 3;
    uint32 stop_from = 4;
    uint32 stop_to = 5;
}

message RouteData {
    int32 idx = 3;
    double weight = 1;
    int32 prev_edge = 2;
}

message RoutesData {
    repeated RouteData data = 1;
}

message Router {
    int32 vertex_count = 1;
    repeated Edge edges = 2;
    repeated EdgeInfo edges_info = 3;
    repeated RoutesData routes_data = 4;
    int32 routes_col_size = 5;
}
